lib/block-supports/layout.php

```diff
@@ -686,8 +686,9 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 	}
 
 	// Add combined layout and block classname for global styles to hook onto.
-	$block_name    = explode( '/', $block['blockName'] );
-	$class_names[] = 'wp-block-' . end( $block_name ) . '-' . $layout_classname;
+	$split_block_name = explode( '/', $block['blockName'] );
+	$full_block_name  = 'core' === $split_block_name[0] ? $split_block_name[1] : implode( '-', $split_block_name );
```

Comment:
Tiny nit: `$split_block_name[1]` assumes the array will have at least two elements. Should we re-use `end()` to account for a custom block registered with the whole name `core`? I think that might wind up being something like:

```suggestion
	$full_block_name  = 'core' === $split_block_name[0] ? end( split_block_name ) : implode( '-', $split_block_name );
```

Reply:
Sure we can do that!

Reply:
I'm not sure it's possible to have more than one `/` in a block name anyway? It'll throw a `Notice: Function WP_Block_Type_Registry::register was called incorrectly.` warning.

Reply:
Hrm, good point! I'd assumed you could register a block without a namespace (so, no `/` character), but it looks like you can't: https://github.com/WordPress/wordpress-develop/blob/5ebe28966e5decbe1862c147bf98db0d8094038e/src/wp-includes/class-wp-block-type-registry.php#L73

I was quite possibly overthinking this! I'm usually quite scared of PHP notices surrounding accessing indexes that aren't available, largely due to my own past mistakes 😅

Reply:
Me too! I think it's fine to leave all the same. `end()` doesn't change the index pointer, but I don't think that's a problem for the rest of the code in this instance.

I was also curious to see whether I could register a custom block with the name `core/something` but it turns out I can't. 🤷🏻